sdk/lib/collection/linked_hash_map.dart

```diff
@@ -6,19 +6,81 @@ part of dart.collection;
 
 /// A hash-table based implementation of [Map].
 ///
-/// The insertion order of keys is remembered,
+/// __The insertion order of keys is remembered__,
 /// and keys are iterated in the order they were inserted into the map.
 /// Values are iterated in their corresponding key's order.
 /// Changing a key's value, when the key is already in the map,
 /// does not change the iteration order,
 /// but removing the key and adding it again
 /// will make it be last in the iteration order.
 ///
+/// **Notice:** Manipulating item count in [forEach] is prohibited. Adding or
+/// deleting items during iteration causes an exception:
+/// [ConcurrentModificationError].
```

Comment:
Same as with `HashMap` PR: 

I don't think "manipulating item count" is quite clear enough here.

There's quite a nice paragraph from the Map documentation, it might be better just copying that?...

It is generally not allowed to modify the map (add or remove keys) while an operation is being performed on the map, for example in functions called during a forEach or putIfAbsent call. Modifying the map while iterating the keys or values may also break the iteration.

Also not sure if specifying the error is actually useful? But I can debate with myself 😁

Like, the message needs to be "don't do this" - it's not an error you'd ever want to catch